viewing_party/party.py

```diff
@@ -0,0 +1,283 @@
+				else: 
+					multiple_highest_genre += f"& {genre} "
+			return multiple_highest_genre
+
+
+# ******WAVE 3********
+
+def get_unique_watched(user_data):
+  
+	# create of 2 lists of dictionaries, one representing movies the user has watched and the other movies that all friends have watched 
+
+	user_watched_list =[]
+	friends_watched_list = []
+
+	if not user_data["watched"]:
+		return user_data["watched"]
+	else:
+		for movie in user_data["watched"]:
+			user_watched_list.append(movie)
+
+	i = 0
+	while i < len(user_data["friends"]):
+		for movie in user_data["friends"][i]["watched"]:
+			friends_watched_list.append(movie)
+		i += 1
+	
+	# convert friend_watched_list and user_watched_list to a list of strings, then to a set of strings, and then find the difference between the 2 sets
+
+	user_watched_set = set()
+	friends_watched_set = set()
+
+	i = 0
+	while i < len(user_watched_list):
+		user_watched_set.add(str(user_watched_list[i]))
+		i += 1
+
+	j = 0
+	while j < len(friends_watched_list):
+		friends_watched_set.add(str(friends_watched_list[j]))
+		j += 1
+
+	user_only_watched_set = user_watched_set - friends_watched_set
+
+
+	# converting difference set back to a list of dictionaries
+
+	import ast
+	user_only_watched_list = list(user_only_watched_set)
+	user_only_watched_list_final = []
+
+	k = 0
+	while k < len(user_only_watched_set):
+		convert_to_dict = ast.literal_eval(user_only_watched_list[k])
+		user_only_watched_list_final.append(convert_to_dict)
+		k += 1
+
+	return user_only_watched_list_final
+
+
+def get_friends_unique_watched(user_data):
+
+	# create of 2 lists of dictionaries, one representing movies the user has watched and the other movies that all friends have watched 
+
+	user_watched_list =[]
+	friends_watched_list = []
+
+	for movie in user_data["watched"]:
+		user_watched_list.append(movie)
+
+	i = 0
+	while i < len(user_data["friends"]):
+		for movie in user_data["friends"][i]["watched"]:
+			friends_watched_list.append(movie)
+		i += 1
+	
+	# convert friend_watched_list and user_watched_list to a list of strings, then to a set of strings, and then find the difference between the 2 sets
+
+	user_watched_set = set()
+	friends_watched_set = set()
+	
+	i = 0
+	while i < len(user_watched_list):
+		user_watched_set.add(str(user_watched_list[i]))
+		i += 1
+
+	j = 0
+	while j < len(friends_watched_list):
+		friends_watched_set.add(str(friends_watched_list[j]))
+		j += 1
+
+	friends_only_watched_set = friends_watched_set - user_watched_set  
+
+	# converting difference set back to a list of dictionaries
+
+	import ast
+	friends_only_watched_list = list(friends_only_watched_set)
+	friends_only_watched_list_final = []
+
+	k = 0
+	while k < len(friends_only_watched_set):
+		convert_to_dict = ast.literal_eval(friends_only_watched_list[k])
+		friends_only_watched_list_final.append(convert_to_dict)
+		k += 1
+	
+	return friends_only_watched_list_final
+
+# *****WAVE 4******
+
+def get_available_recs(user_data):
```

Comment:
this works well and also think about how you could use `get_friends_unique_watched(user_data)` here